### ISSUE TITLE
Fail on loading LAMMPS Data file with non-empty blank lines

### ISSUE THREAD

**Issue by DmitriyMarin** [chemfiles/chemfiles#485]
The loader for the **LAMMPS Data format** fails when loading LAMMPS Data files (e.g., *.lmp) that contain blank lines with whitespaces because it expects the blank lines to be empty. I am not sure whether the [LAMMPS Data format](https://docs.lammps.org/2001/data_format.html) actually specifies that blank lines should be empty or whether they are allowed to contain whitespaces. 

Here is a LAMMPS Data file to reproduce the issue (a line before the *Atoms* section contains some spaces):

```
# LAMMPS data file

1 atoms
1 atom types

0.00 10.0000000000 xlo xhi 
0.00 10.0000000000 ylo yhi 
0.00 10.0000000000 zlo zhi 
         
Atoms  # full

1 1 1 0.0 5.0 5.0 5.0

```

Here is a simple Python code to reproduce:
```python
from chemfiles import Trajectory, Frame

trajectory = Trajectory(path = "file.lmp", format = "LAMMPS Data")
try:
    frame = trajectory.read()
except:
    print('Exception')

print(f"There are {len(frame.atoms)} atoms in the frame")
```

This code will simply fail without raising an exception (due to `assert`s). This also means that the **chemfiles** Python package has been built without `NDEBUG`.

I understand, that LAMMPS Data files might be should not have non-empty blank lines. But this also raises the following question: would it be better in this case to throw an exception instead of using `assert`? If **chemfiles** is built with `NDEBUG` it won't fail but won't provide any information on why it wasn't able to load.

**Comment by Luthaf** [comment]
Thanks for the report! This does look like an error on chemfiles side and not the file formatting, I'll fix it.

Regarding the `NDEBUG` question, which OS are you running on? For me this code is an infinite loop and does not hit any assertion.

**Comment by Luthaf** [comment]
I don't quite understand how we end up compiling code with `-UNDEBUG`. We build the weels on PyPI with [`cibuildwheel`](https://github.com/pypa/cibuildwheel/), which does not seems to set anything special (see the log here: https://github.com/chemfiles/chemfiles.py/actions/runs/6301749249/job/17107468253?pr=52). 

And we set the build mode to `Release` here anyway: https://github.com/chemfiles/chemfiles.py/blob/839ac750c2b2618353779bff9e07aa2e90dd9ac6/setup.py#L46.

What's the exact output/error you get with the version from PyPI/pip?

**Comment by DmitriyMarin** [comment]
I am not sure whether the Python package is actually compiled without `NDEBUG` (or with `UNDEBUG`), but the execution of the above-mentioned example simply exits the interpreter without any output or error, for example:

```shell
PS D:\Samples\tests\lammps> python
Python 3.11.1 (tags/v3.11.1:a7a450f, Dec  6 2022, 19:58:39) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from chemfiles import Trajectory, Frame
>>>
>>> trajectory = Trajectory(path = "test_with_nonempty_blank_line.lmp", format = "LAMMPS Data")
>>> try:
...     frame = trajectory.read()
... except:
...     print('Exception')
...
PS D:\Samples\tests\lammps>
```


`pip freeze` gives me `chemfiles==0.10.4`

**Comment by Luthaf** [comment]
This is now merged on master, I'll update the release 0.10 branch with the change & try to do a patch release ASAP.

**Comment by DmitriyMarin** [comment]
Great! Thanks!
